lib/ansible/modules/cloud/vagrant/vagrant_box.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# Copyright (c), Chafik Belhaoues  <chafik.bel@gmail.com>, 2017
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: vagrant_box
+short_description: Manage Vagrant boxes
+version_added: "2.5"
+author: "Chafik Belhaoues (@africanzoe)"
+description:
+   - This module can add|remove|update a Vagrant box on a target host.
+options:
+    box:
+      description:
+        - The name of the box to be managed, it could be a path to a file, URL or shorthand name from Vagrant cloud.
+      aliases: ['path', 'url']
+      required: true
+      default: None
+    state:
+      description:
+        - Indicate the desired state of the box
+      choices: ['present', 'absent', 'updated']
+      required: false
+      default: present
+    force:
+      description:
+        - Specify wether to force the add or remove actions.
+      required: false
+      default: false
+    version:
+      description:
+        - Specify the version of the managed box.
+        - Please consider using an explicit version with the state "absent".
+      required: false
+      default: latest
+    provider:
+      description:
+        - Specify the provider for the box.
+      choices: ['virtualbox', 'libvirt', 'hyperv', 'vmware_desktop']
+      required: false
+      default: "virtualbox"
+    insecure:
+      description:
+        - Indicate that the SSL certificates will not be verified if the URL is an HTTPS URL.
+      required: false
+      default: false
+    trust_location:
+      description:
+        - Indicate wether to trust 'Location' header from HTTP redirects and use the same credentials for subsequent urls as for the initial one or not.
+      required: false
+      default: false
+    cacert:
+      description:
+        - The path on the target host to the FILE that contains CA certificate for SSL download.
+      required: false
+      default: false
+    capath:
+      description:
+        - The path on the target host to the DIRECTORY for the CA used to verify the peer.
+      required: false
+      default: false
+    cert:
+      description:
+        - The path on the target host to the a client certificate to use when downloading the box, if necessary.
+      required: false
+      default: false
+    name:
+      description:
+        - The name to give to the box if added from a file.
+        - When adding a box from a catalog, the name is included in the catalog entry and should not be specified.
+      required: false
+      default: false
+    checksum:
+      description:
+        - The checksum for the box that is downloaded.
+      required: false
```

Comment:
Required is False by-default, so no need to describe explicitly in docs.

Reply:
Done.